crates/storage/src/entities.rs

```diff
@@ -644,17 +644,19 @@ impl Element {
     /// # Parameters
     ///
     /// * `path` - The path to the [`Element`] in the hierarchy of the storage.
+    /// * `id` - The id of the [`Element`] in the hierarchy of the storage.
     ///
     /// # Panics
     ///
     /// This method can technically panic if the system time goes backwards, to
     /// before the Unix epoch, which should never ever happen!
     ///
     #[must_use]
-    pub fn new(path: &Path) -> Self {
+    pub fn new(path: &Path, id: Option<Id>) -> Self {
```

Comment:
I see we are extensively using id = `None`, would it be better to create another constructor?

Reply:
Its legacy code from Dan which has to be removed, but in next PRs.